node/lib/cmd/commit.js

```diff
@@ -144,9 +158,118 @@ ${colors.yellow("Please stage changes in new submodules before committing.")}\
     }
 
     if ("" === args.message) {
-        console.error("Aborting commit due to empty commit message.");
-        process.exit(1);
+        abortForNoMessage();
     }
 
     yield Commit.commit(repo, args.all, repoStatus, args.message);
 });
+
+const doAmend = co.wrap(function *(args) {
+    const colors = require("colors");
+    const path   = require("path");
+
+    const Commit          = require("../util/commit");
+    const GitUtil         = require("../util/git_util");
+    const StatusUtil      = require("../util/status_util");
+    const SubmoduleUtil   = require("../util/submodule_util");
+    const UserError       = require("../util/user_error");
+
+    const repo = yield GitUtil.getCurrentRepo();
+    const cwd = process.cwd();
+    const workdir = repo.workdir();
+    const relCwd = path.relative(workdir, cwd);
+    let status = yield StatusUtil.getRepoStatus(repo, {
+        showMetaChanges: args.meta,
+        includeClosedSubmodules: true,
+    });
+    const head = yield repo.getHeadCommit();
+
+    // Load up the set of submodules in existence on the previous commit, if
+    // any.
+
+    let oldSubs = {};
+    const parent = yield GitUtil.getParentCommit(repo, head);
+    if (null !== parent) {
+        oldSubs = yield SubmoduleUtil.getSubmodulesForCommit(repo, parent);
+    }
+
+    // Check to see if the repo is in a valid state to be amended.
+
+    const amendable = yield Commit.checkIfRepoIsAmendable(repo,
+                                                          status,
+                                                          oldSubs);
+    let bad = "";
+    amendable.newCommits.forEach(sub => {
+        bad += `${colors.red(sub)} has new commits.\n`;
+    });
+    amendable.mismatchCommits.forEach(sub => {
+        bad += `The commit for ${colors.red(sub)} does not match.`;
+    });
+    if ("" !== bad) {
+        throw new UserError("Cannot make amend commit:\n" + bad);
+    }
+
+    // May have opened repos, so we need to update 'status'.
+
+    status = amendable.status;
+
+    // Calculate the changes that will be made by this amend -- so we know:
+    // (a) which subs to amend
+    // (b) what changes to stage (if any)
+    // (c) if an editor prompt is required, what information to display
+
+    const amendChanges = yield Commit.getAmendChanges(repo,
+                                                      oldSubs,
+                                                      status,
+                                                      args.meta,
+                                                      args.all);
+
+    // Update 'status' to reflect changes to be applied with amend.
+
+    status = amendChanges.status;
+
+    // If no message, use editor.
+
+    if (null === args.message) {
+        const headSig = head.author();
+        const time = headSig.when();
+        const date = new Date(time.time() * 1000);
```

Comment:
This is just the date we show in the message in the editor. The actual date saved is done by the `NodeGit.Commit.amend` method, which uses the original date that we show here.